GreetingTest.java

```diff
@@ -0,0 +1,23 @@
+import static org.junit.Assert.assertEquals;
+import org.junit.*;
+import java.io.*;
+
+public class GreetingTest{
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();    
+
+    @Before
+    public void setUpStreams() {
+        System.setOut(new PrintStream(outContent));
+    }
+    
+    @After
+    public void cleanUpStreams() {
+        System.setOut(null);
```

Comment:
Effectively "breaks" System.out for any subsequent use.

Each test should prepare whatever they need, then do their thing, then restore everything back to the way it was  prior to the test.